dbt_py/main.py

```diff
@@ -64,4 +64,5 @@ def main() -> None:
     - https://docs.getdbt.com/reference/programmatic-invocations
     """
     dbt.context.base.get_context_modules = new_get_context_modules
-    dbt.cli.main.dbtRunner().invoke(sys.argv[1:])
+    result = dbt.cli.main.dbtRunner().invoke(sys.argv[1:])
+    return not result.success
```

Comment:
Thanks for catching this and raising the PR!

I'm just trying to understand the change -- have you confirmed that adding the `return` would push the correct exit code?

Looking at [the dbt docs for the programmatic invocation](https://docs.getdbt.com/reference/programmatic-invocations#dbtrunnerresult), I was expecting something like:

```suggestion
    if result.success:
        sys.exit(0)
    elif result.exception is None:
        sys.exit(1)
    else:
        sys.exit(2)
```

Given that people are actually using this repo, I'll also get the integration tests fixed this afternoon

Reply:
Thanks - I was mainly looking for the failure case in general. The setup you've got is more correct, my fix was just a quick one I did for a CI pipeline. Glad to see there's proper guidance on how to deal with it